src/JustSaying/Messaging/Middleware/Handle/HandlerInvocationMiddleware.cs

```diff
@@ -8,10 +8,8 @@ namespace JustSaying.Messaging.Middleware;
 /// This middleware is responsible for resolving a message handler and calling it.
 /// </summary>
 /// <typeparam name="T">The type of the message that the message handler handles.</typeparam>
-public sealed class HandlerInvocationMiddleware<T>(Func<HandlerResolutionContext, IHandlerAsync<T>> handlerResolver) : MiddlewareBase<HandleMessageContext, bool> where T : Message
+public sealed class HandlerInvocationMiddleware<T>(IHandlerResolver handlerResolver) : MiddlewareBase<HandleMessageContext, bool> where T : Message
```

Comment:
Just skim reading at this point rather than _properly_ reviewing the change (I'll do that later), but this would be a binary breaking change (as would the other changes to any `public` member), and I don't think we'd want to have to rev to v8 right now.

This would need doing in a way that is opt-in additional functionality, not a change to the existing API surface.

Reply:
Thanks for alerting me to this!

Correct me if I'm wrong, but my latest commit will resolve this issue - we don't change existing public methods, and leave in place the old public classes, and instead mark them obsolete.

We then use new methods and classes for the newer functionality